frontend/src/Pages/BMLogHoursPage/BMLogHoursPage.js

```diff
@@ -0,0 +1,120 @@
+import "./BMLogHoursPage.css";
+import { useState } from "react";
+import { Form, Button } from "react-bootstrap";
+
+function BMLogHoursPage(props) {
+    //stores inputs from form
+    const [startDT, setStartDT] = useState("");
+    const [endDT, setEndDT] = useState("");
+    const [tasks, setTasks] = useState("");
+
+    var opportunities;
+    var key;
+
+    if (props.user !== null && props.user.opportunities !== null) {
+        opportunities = props.user.opportunities;
+        Object.keys(opportunities).map((item) => {
+            key = item;
+            return null;
+        })
+        opportunities = opportunities[key];
+    }
+
+    function isValidForm() {
+        if (startDT === "" || endDT === "" || tasks === "") {
+            alert("Incomplete form. Please try again.");
+            return false;
+        }
+        else if (startDT > endDT) {
+            alert("Start and end dates are invalid. Please enter dates again.");
+            return false;
+        }
+        return true;
+    }
+
+    //push form as element in opportunity array in volunteer schema
+    function handleSubmit() {
+        // make sure opportunties is defined
+        if (opportunities === null || opportunities === undefined) return;
```

Comment:
I am a little confused by this line. Maybe instead make a default Board Member opportunity in our database and add it to their opportunity array if the user is a board member, then append each of their logged hours to the task array for that opportunity? Since with this if statement, the first time a user tries to add board member hours to their user it will not let them because there is no way to register for a board member opportunity. Or am I missing how your logic is working here?

Reply:
Yeah, I'm thinking this line can stay since it's a good check, but maybe just append them to this opportunity id: 6099c78c001ee300081c1dab as a task and then change it so on registration if they check the board member true button on registration, append this opportunity to their array with zero tasks. We can also filter this opportunity out from the usual opportunity display